### PR TITLE
Docs/atrujill0 1.9 review

### PR DESCRIPTION
**Pull request by trujillo-adam**
This PR addresses grammar and writing style issues in the `templatestring` documentation for 1.9.

### CODE REVIEW

File: website/docs/language/functions/templatestring.mdx [hunk 1]
```diff
@@ -4,39 +4,31 @@ description: |-
...
--> **Note:** The `templatestring` function is intended for advanced use cases. Most use cases require only a [string template expression](/terraform/language/expressions/strings#string-templates). To render a template from a file, use the [`templatefile` function](/terraform/language/functions/templatefile).
+This topic provides reference information about the `templatestring` function. The `templatestring` function renders a string defined elsewhere in the module as a template using a set of variables. 
 
-`templatestring` renders a template given as a string value, using a supplied set of template variables.
+## Introduction
 
-```hcl
-templatestring(ref, vars)
-```
+The primary use case for the `templatestring` function is to render templates fetched as a single string from remote locations using data resources. The function enables advanced use cases where a [string template expression](/terraform/language/expressions/strings#string-templates) is insufficient, such as when the template is available from a named object declared in the current module. To render a template from a file, use the [`templatefile` function](/terraform/language/functions/templatefile).
```
Comment (BrianMMcClain) [inline website/docs/language/functions/templatestring.mdx:13]: This reads to me that you can _only_ use a data source to get the template string. At least technically speaking, it looks like there's no limitation on local values, resource attributes, or even other variables. Are we advising that users _only_ use data sources with the `templatestring` function?

**Review by BrianMMcClain — APPROVED** [review]
Just a couple thoughts from my end, thanks!

File: website/docs/language/functions/templatestring.mdx (hunk 1 above)
Comment (trujillo-adam) [reply website/docs/language/functions/templatestring.mdx:13]: I don't think that is the intent. This is an edit of the original version. I'll try a different wording.

Comment (trujillo-adam) [inline website/docs/language/functions/templatestring.mdx:13]:
```suggestion
The primary use case for the `templatestring` function is to render templates fetched as a single string from remote locations. The function enables advanced use cases where a [string template expression](/terraform/language/expressions/strings#string-templates) is insufficient, such as when the template is available from a named object declared in the current module. To render a template from a file, use the [`templatefile` function](/terraform/language/functions/templatefile).
```
I don't think that the phrase about data resources was actually adding value.

**Comment by apparentlymart** [comment]
I trust your judgement here and so I don't have any specific feedback on the changes you've made. If this is more consistent with our current documentation standards then that's most important.

The only note I wanted to add is that for this function in particular I had some unusual author intent in that I was trying to speak both to people who are trying to use this function for its intended purpose _and_ to people who are new to Terraform, haven't yet learned that the Terraform language has [built-in template support](https://developer.hashicorp.com/terraform/language/expressions/strings#string-templates), and so might get stuck trying to figure out how to make this function work when really they shouldn't be using this function at all.

This function was a long time coming because we already had the prior experience of people misunderstanding what [the `template_file` data source](https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file) was intended for, and in particular would try to use it with templates written inline and would then get confusing error messages because Terraform can't understand what the author was intending.

I usually keep this sort of thing vague because I don't want to give the impression that I'm calling specific individuals out for using Terraform incorrectly, but I think in this case it's probably better to be specific to be clearer about where I'm coming from, so here's two specific examples:

- [Using an inline template directly](https://stackoverflow.com/questions/52522996/correct-way-to-terraform-template-provider-data-source-output)
- [Using inline template fragments as part of a more complex expression](https://stackoverflow.com/questions/76902697/updating-to-built-in-templatefile-in-terraform-from-using-template-file-in-d/76902788#76902788)

(I want to reassure the folks who wrote those questions -- in case they see this at some point -- that I don't believe they wrong to be confused by Terraform's behavior here. That Terraform feature was designed poorly and so was unable to give good feedback.)

Both of those situations and others like it intentionally fail for `templatefile`. The function implementation tries to give _some_ feedback about the intended usage and alternatives in that case, but there's only so much context we can reasonably pack into an error message. So I think an important target audience for this function's documentation is someone who already saw such an error message and is hoping to find more context about what it means and what exactly they might do instead.

The prior content of this page was my initial attempt to speak to that reader without it overwhelming the page, but I wasn't very happy with it so I'm glad you have time to revise it. Again, I trust your instincts and so I'm satisfied with whatever you think is best, taking this special context into account.

**Comment by trujillo-adam** [comment]
Discussed with @apparentlymart offline and concluded that this revision is acceptable.

**Pull request merged**